src/scan-iterator.ts

```diff
@@ -153,10 +127,8 @@ class ScanIterator<V> implements AsyncIterableIterator<V> {
       opts,
       receiver,
     };
-    await this._invoke('scan', args);
-    if (responseItems.length !== scanPageSize) {
-      this._moreItemsToLoad = false;
-    }
+    this._loadPromise = this._invoke('scan', args);
```

Comment:
Can remove `limit` from rpc.

Reply:
Yes, but it is more efficient to keep it.

We could change the callback to return a boolean to determine if we want to continue but I don't want to go down that route with workers on the horizon.